File: ios/WeightTracker/Screens/ExerciseScreen.swift
import SwiftUI

let EXERCISE_TYPES: [(type: String, name: String, icon: String, caloriesPerMin: Int)] = [
    ("running", "跑步", "🏃", 10),
    ("swimming", "游泳", "🏊", 8),
    ("cycling", "骑行", "🚴", 7),
    ("gym", "健身", "🏋️", 6),
    ("yoga", "瑜伽", "🧘", 3),
    ("walking", "步行", "🚶", 4),
    ("jumping", "跳绳", "⏫", 12),
    ("hiit", "HIIT", "⚡", 14)
]

struct ExerciseScreen: View {
    @ObservedObject var viewModel: WeightViewModel
    @State private var showAddSheet = false
    @State private var selectedType = EXERCISE_TYPES[0]
    @State private var duration = ""
    @State private var selectedDate = Date()
    
    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                // Today Stats
                VStack(spacing: 12) {
                    HStack {
                        Text("🏃 今日运动")
                            .font(.headline)
                        Spacer()
                    }
                    
                    HStack(spacing: 20) {
                        VStack {
                            Text("\(viewModel.todayExerciseCalories)")
                                .font(.title)
                                .fontWeight(.bold)
                                .foregroundColor(.green)
                            Text("消耗 kcal")
                                .font(.caption)
                                .foregroundColor(.secondary)
                        }
                    }
                    .frame(maxWidth: .infinity)
                }
                .padding()
                .background(Color.green.opacity(0.1))
                .cornerRadius(12)
                
                // Add Button
                Button(action: { showAddSheet = true }) {
                    Label("记录运动", systemImage: "plus.circle.fill")
                        .font(.headline)
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .padding()
                        .background(
                            LinearGradient(
                                colors: [.green, .green.opacity(0.8)],
                                startPoint: .leading,
                                endPoint: .trailing
                            )
                        )
                        .cornerRadius(12)
                }
                
                // Exercise History
                VStack(alignment: .leading, spacing: 8) {
                    Text("📋 运动历史")
                        .font(.headline)
                    
                    if viewModel.exercises.isEmpty {
                        Text("暂无运动记录")
                            .foregroundColor(.secondary)
                            .frame(maxWidth: .infinity)
                            .padding()
                    } else {
                        ForEach(viewModel.exercises.suffix(20).reversed()) { exercise in
                            HStack {
                                Text(exercise.typeIcon)
                                    .font(.title)
                                
                                VStack(alignment: .leading) {
                                    Text(exercise.typeName)
                                        .font(.headline)
                                    Text("\(exercise.date) · \(exercise.duration)分钟")
                                        .font(.caption)
                                        .foregroundColor(.secondary)
                                }
                                
                                Spacer()
                                
                                Text("\(exercise.calories) kcal")
                                    .foregroundColor(.orange)
                                    .fontWeight(.bold)
                            }
                            .padding(.vertical, 8)
                            Divider()
                        }
                    }
                }
                .padding()
                .background(Color(.systemBackground))
                .cornerRadius(12)
                .shadow(color: .black.opacity(0.1), radius: 5)
            }
            .padding()
        }
        .background(Color(.systemGroupedBackground))
        .navigationTitle("🏃 运动记录")
        .sheet(isPresented: $showAddSheet) {
            AddExerciseSheet(
                viewModel: viewModel,
                selectedType: $selectedType,
                duration: $duration,
                selectedDate: $selectedDate
            )
        }
    }
}

struct AddExerciseSheet: View {
    @ObservedObject var viewModel: WeightViewModel
    @Binding var selectedType: (type: String, name: String, icon: String, caloriesPerMin: Int)
    @Binding var duration: String
    @Binding var selectedDate: Date
    @Environment(\.dismiss) var dismiss
    
    var calculatedCalories: Int {
        guard let mins = Int(duration) else { return 0 }
        return mins * selectedType.caloriesPerMin
    }
    
    var body: some View {
        NavigationStack {
            Form {
                DatePicker("日期", selection: $selectedDate, displayedComponents: .date)
                
                Section("运动类型") {
                    LazyVGrid(columns: [GridItem(.flexible()), GridItem(.flexible()), GridItem(.flexible()), GridItem(.flexible())], spacing: 8) {
                        ForEach(EXERCISE_TYPES, id: \.type) { type in
                            Button(action: { selectedType = type }) {
                                VStack {
                                    Text(type.icon)
                                        .font(.title)
                                    Text(type.name)
                                        .font(.caption)
                                }
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(selectedType.type == type.type ? Color.purple.opacity(0.2) : Color.clear)
                                .cornerRadius(8)
                            }
                            .buttonStyle(.plain)
                        }
                    }
                }
                
                HStack {
                    Text("运动时长")
                    TextField("分钟", text: $duration)
                        .keyboardType(.numberPad)
                        .multilineTextAlignment(.trailing)
                    Text("分钟")
                        .foregroundColor(.secondary)
                }
                
                if !duration.isEmpty {
                    HStack {
                        Text("预计消耗")
                        Spacer()
                        Text("\(calculatedCalories) kcal")
                            .fontWeight(.bold)
                            .foregroundColor(.orange)
                    }
                }
            }
            .navigationTitle("记录运动")
            .toolbar {
                ToolbarItem(placement: .cancellationAction) {
                    Button("取消") { dismiss() }
                }
                ToolbarItem(placement: .confirmationAction) {
                    Button("保存") {
                        guard let mins = Int(duration), mins > 0 else { return }
                        let dateFormatter = DateFormatter()
                        dateFormatter.dateFormat = "yyyy-MM-dd"
                        viewModel.addExercise(
                            date: dateFormatter.string(from: selectedDate),
                            type: selectedType.type,
                            typeName: selectedType.name,
                            typeIcon: selectedType.icon,
                            duration: mins,
                            calories: calculatedCalories
                        )
                        duration = ""
                        dismiss()
                    }
                    .disabled(duration.isEmpty)
                }
            }
        }
    }
}

File: flutter/lib/screens/exercise_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:intl/intl.dart';
import '../providers/app_provider.dart';
import '../models/models.dart';

const _exerciseTypes = [
  ('running', '跑步 🏃', 10),
  ('swimming', '游泳 🏊', 8),
  ('cycling', '骑行 🚴', 7),
  ('gym', '健身 🏋️', 6),
  ('yoga', '瑜伽 🧘', 3),
  ('walking', '步行 🚶', 4),
  ('jumping', '跳绳 ⏫', 12),
  ('hiit', 'HIIT ⚡', 14),
];

class ExerciseScreen extends StatefulWidget {
  const ExerciseScreen({super.key});

  @override
  State<ExerciseScreen> createState() => _ExerciseScreenState();
}

class _ExerciseScreenState extends State<ExerciseScreen> {
  String _selectedType = 'running';
  final _durationController = TextEditingController();
  late String _selectedDate;

  @override
  void initState() {
    super.initState();
    _selectedDate = DateFormat('yyyy-MM-dd').format(DateTime.now());
  }

  @override
  void dispose() {
    _durationController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('🏃 运动记录'),
        centerTitle: true,
      ),
      body: Consumer<AppProvider>(
        builder: (context, provider, child) {
          return SingleChildScrollView(
            padding: const EdgeInsets.all(16),
            child: Column(
              children: [
                // Today Stats
                Card(
                  color: Colors.green.withOpacity(0.1),
                  child: Padding(
                    padding: const EdgeInsets.all(16),
                    child: Column(
                      children: [
                        const Text(
                          '🏃 今日运动',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        const SizedBox(height: 12),
                        Text(
                          '${provider.todayExerciseCalories}',
                          style: const TextStyle(
                            fontSize: 36,
                            fontWeight: FontWeight.bold,
                            color: Colors.green,
                          ),
                        ),
                        const Text('消耗 kcal'),
                      ],
                    ),
                  ),
                ),
                const SizedBox(height: 16),

                // Add Exercise Button
                SizedBox(
                  width: double.infinity,
                  child: FilledButton.icon(
                    onPressed: () => _showAddDialog(context, provider),
                    icon: const Icon(Icons.add),
                    label: const Text('记录运动'),
                  ),
                ),
                const SizedBox(height: 16),

                // History
                Card(
                  child: Padding(
                    padding: const EdgeInsets.all(16),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        const Text(
                          '📋 运动历史',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        const SizedBox(height: 8),
                        if (provider.exercises.isEmpty)
                          const Center(
                            child: Padding(
                              padding: EdgeInsets.all(16),
                              child: Text('暂无运动记录'),
                            ),
                          )
                        else
                          ...provider.exercises.reversed.take(20).map(
                            (exercise) => ListTile(
                              contentPadding: EdgeInsets.zero,
                              leading: Text(
                                exercise.typeIcon,
                                style: const TextStyle(fontSize: 24),
                              ),
                              title: Text(exercise.typeName),
                              subtitle: Text('${exercise.date} · ${exercise.duration}分钟'),
                              trailing: Text(
                                '${exercise.calories} kcal',
                                style: const TextStyle(
                                  fontWeight: FontWeight.bold,
                                  color: Colors.orange,
                                ),
                              ),
                            ),
                          ),
                      ],
                    ),
                  ),
                ),
              ],
            ),
          );
        },
      ),
    );
  }

  void _showAddDialog(BuildContext context, AppProvider provider) {
    _selectedType = 'running';
    _durationController.clear();
    _selectedDate = DateFormat('yyyy-MM-dd').format(DateTime.now());

    showDialog(
      context: context,
      builder: (context) => StatefulBuilder(
        builder: (context, setState) {
          final selectedExercise = _exerciseTypes.firstWhere(
            (e) => e.$1 == _selectedType,
            orElse: () => _exerciseTypes[0],
          );
          final duration = int.tryParse(_durationController.text) ?? 0;
          final calories = duration * selectedExercise.$3;

          return AlertDialog(
            title: const Text('记录运动'),
            content: SingleChildScrollView(
              child: Column(
                mainAxisSize: MainAxisSize.min,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  TextField(
                    decoration: const InputDecoration(
                      labelText: '日期',
                      border: OutlineInputBorder(),
                    ),
                    controller: TextEditingController(text: _selectedDate),
                    onChanged: (v) => _selectedDate = v,
                  ),
                  const SizedBox(height: 12),
                  const Text('运动类型'),
                  const SizedBox(height: 8),
                  Wrap(
                    spacing: 8,
                    runSpacing: 8,
                    children: _exerciseTypes.map((type) {
                      final isSelected = _selectedType == type.$1;
                      return FilterChip(
                        label: Text(type.$2),
                        selected: isSelected,
                        onSelected: (_) => setState(() => _selectedType = type.$1),
                      );
                    }).toList(),
                  ),
                  const SizedBox(height: 12),
                  TextField(
                    controller: _durationController,
                    decoration: const InputDecoration(
                      labelText: '运动时长 (分钟)',
                      border: OutlineInputBorder(),
                    ),
                    keyboardType: TextInputType.number,
                    onChanged: (_) => setState(() {}),
                  ),
                  if (duration > 0) ...[
                    const SizedBox(height: 8),
                    Text('预计消耗: $calories kcal'),
                  ],
                ],
              ),
            ),
            actions: [
              TextButton(
                onPressed: () => Navigator.pop(context),
                child: const Text('取消'),
              ),
              FilledButton(
                onPressed: duration > 0
                    ? () {
                        provider.addExercise(ExerciseRecord(
                          date: _selectedDate,
                          type: selectedExercise.$1,
                          typeName: selectedExercise.$2.split(' ').first,
                          typeIcon: selectedExercise.$2.split(' ').last,
                          duration: duration,
                          calories: calories,
                        ));
                        Navigator.pop(context);
                      }
                    : null,
                child: const Text('保存'),
              ),
            ],
          );
        },
      ),
    );
  }
}
